evap/results/exporters.py

```diff
@@ -94,37 +94,51 @@ def export(self, response, selection_list, include_not_enough_voters=False, incl
                 evaluation_states.extend(['evaluated', 'reviewed'])
 
             used_questionnaires = set()
-            for evaluation in self.semester.evaluations.filter(
-                state__in=evaluation_states, course__degrees__in=degrees, course__type__in=course_types
-            ).distinct():
+            evaluations_filter = Q(course__semester__in=semesters, state__in=evaluation_states, course__degrees__in=degrees, course__type__in=course_types)
+            if contributor:
+                evaluations_filter = evaluations_filter & (Q(course__responsibles__in=[contributor]) | Q(contributions__contributor__in=[contributor]))
+            evaluations = Evaluation.objects.filter(evaluations_filter).distinct()
+            for evaluation in evaluations:
                 if evaluation.is_single_result:
                     continue
                 if not evaluation.can_publish_rating_results and not include_not_enough_voters:
                     continue
                 results = OrderedDict()
-                for questionnaire_result in collect_results(evaluation).questionnaire_results:
-                    if all(not question_result.question.is_rating_question or question_result.counts is None for question_result in questionnaire_result.question_results):
-                        continue
-                    results.setdefault(questionnaire_result.questionnaire.id, []).extend(questionnaire_result.question_results)
-                    used_questionnaires.add(questionnaire_result.questionnaire)
+                for contribution_result in collect_results(evaluation).contribution_results:
+                    for questionnaire_result in contribution_result.questionnaire_results:
+                        if all(not question_result.question.is_rating_question or question_result.counts is None for question_result in questionnaire_result.question_results):
+                            continue
+                        if not contributor or contribution_result.contributor is None or contribution_result.contributor == contributor:
+                            results.setdefault(questionnaire_result.questionnaire.id, []).extend(questionnaire_result.question_results)
+                            used_questionnaires.add(questionnaire_result.questionnaire)
```

Comment:
You could inverse the control flow here (just as is done above), it would allow shortening the condition:
```python
if contributor and contribution_result.contributor != contributor:
    continue

results.setdefault(...)
used_questionnaired.add(...)
```

Reply:
we still need the check for `contribution_result.contributor is None` so that general contributions are not skipped.
the negation would be `contributor and contribution_result.contributor is not None and contribution_result.contributor != contributor`

Reply:
Ah true, sorry.